### PR TITLE
[NO ISSUE] Update docs to be more PSO 6.0 friendly

### PR DESCRIPTION
**Pull request by sdodsley**
Remove all unnecessary references to k8s and PSO versions not supported by PSO 6
Consolidate READMEs into something more readable.

### CODE REVIEW

File: docs/csi-snapshot-clones.md [hunk 1]
```diff
@@ -14,30 +14,11 @@ These features use native Kubernetes APIs to call the feature-set in the underly
...
-```
-kubectl describe -n kube-system pod kube-api-server-<node name> | grep feature-gates
-```
-
-This should result is the following if the feature gates are correctly set.
-
-```
- --feature-gates=VolumeSnapshotDataSource=True,VolumePVCDataSource=True
-```
+* For the snapshot feature PSO does not install the Snapshot CRDs by default through Helm. Please ensure the Snapshot CRDs and Controller by your Kubernetes deployment. If they aren't the refer to the [Kuberenetes CSI Snapshot documentation](https://kubernetes-csi.github.io/docs/snapshot-controller.html) for installation details.
```
Comment (dsupure) [inline docs/csi-snapshot-clones.md:17]: If they aren't `then`

File: pureStorageDriver/README.md [hunk 2]
```diff
@@ -1,48 +1,12 @@
...
-1. Ensure the feature gate is enabled via the following Kubernetes feature flag: ```--feature-gates=CSIDriverRegistry=true```
-2. Either ensure the CSIDriver CRD is installed cluster with the following command:
-
-```bash
-kubectl create -f https://raw.githubusercontent.com/kubernetes/csi-api/master/pkg/crd/manifests/csidriver.yaml
-```
+1. Pure Service Orchestrator deploys a CockroachDB datastore replicated across the provided storage backends. **Note: All licensing issues regarding this are covered by Pure Storage**
+2. Currently, there is **no upgrade supported** from previous versions that do not deploy the datastore.
+3. You **MUST** supply a unique `clusterID` in values.yaml. This was previously called `namespace.pure`. `clusterID` must be less than or equal to 22 characters in length. `clusterID` must be unique between **all** Kubernetes clusters using your Pure devices or naming conflicts will result.
+4. `helm uninstall` will perform the initial uninstallation, but some pods will continue to clean up post-installation. They should go away after cleanup is complete.
```
Comment (dsupure) [inline pureStorageDriver/README.md:9]: Can you also add the 5th bullet point mentioned that users **MUST** install the NTP service to make the db work?
I would expect many users will report that PSO fail to work due to this issue.

File: README.md [hunk 3]
```diff
@@ -25,12 +25,18 @@ To ensure your services stay robust, PSO self-heals – so you’re protected ag
...
-  - CoreOS (Ladybug 1298.6.0 and above)
+  - Red Hat CoreOS 4.4+
   - RHEL 7
   - Ubuntu 16.04
   - Ubuntu 18.04
+  - Ubuntu 20.04
 - #### Environments Supported*:
-  - Refer to the README for the type of PSO installation required
+  - Kubernetes 1.17+
+    - [Note: For version less tan 1.17.6/1.18.6 please refer to this issue using vxlan with Flannel or Calico](https://github.com/kubernetes/kubernetes/issues/87852)
```
[the remark this answers is not in the document]
Comment (sdodsley) [reply README.md]: Done

File: docs/csi-snapshot-clones.md (hunk 1 above)
Comment (sdodsley) [reply docs/csi-snapshot-clones.md]: Done

File: pureStorageDriver/README.md (hunk 2 above)
Comment (sdodsley) [reply pureStorageDriver/README.md:9]: Done

File: pureStorageDriver/README.md [hunk 4]
```diff
@@ -170,67 +190,26 @@ volumesnapshotcontents.snapshot.storage.k8s.io   2019-11-21T17:25:23Z
 volumesnapshots.snapshot.storage.k8s.io          2019-11-21T17:25:23Z
 ```
```
Comment (carysyd) [inline pureStorageDriver/README.md:192]: I think we should have a section here to direct user to "https://kubernetes-csi.github.io/docs/snapshot-controller.html" if they don't see these CRDs in their system - which is not installed by default anyways.

Comment (sdodsley) [reply pureStorageDriver/README.md:192]: Will add.

File: pureStorageDriver/README.md [hunk 5]
```diff
@@ -1,48 +1,14 @@
...
-
-```bash
-kubectl create -f https://raw.githubusercontent.com/kubernetes/csi-api/master/pkg/crd/manifests/csidriver.yaml
-```
+1. Pure Service Orchestrator deploys a CockroachDB datastore replicated across the provided storage backends. **Note: All licensing issues regarding this are covered by Pure Storage**
+2. Currently, there is **no upgrade supported** from previous versions that do not deploy the datastore.
+3. You **MUST** supply a unique `clusterID` in values.yaml. This was previously called `namespace.pure`. `clusterID` must be less than or equal to 22 characters in length. `clusterID` must be unique between **all** Kubernetes clusters using your Pure devices or naming conflicts will result.
+4. `helm uninstall` will perform the initial uninstallation, but some pods will continue to clean up post-installation. They should go away after cleanup is complete.
+5. Note that PSO CSI only supports the Beta version snapshotter APIs. The snapshotter CRDs for the Beta version APIs have been upgraded, therefore use only release-2.0 CRDs as detailed below.
+5. An implementation of the Network Time Protocol **MUST** be running on all nodes in the Kubernetes cluster.
```
Comment (pure-jliao) [inline pureStorageDriver/README.md:11]: nit: `6.`

File: pureStorageDriver/README.md [hunk 6]
```diff
@@ -60,31 +26,78 @@ More details on using Read-Write-Many (RWX) volumes with Kubernetes can be found
...
+```
+
+### Configure Helm
+
+Add the Pure Storage PSO helm repository to your helm installation.
 
 ```bash
 helm repo add pure https://purestorage.github.io/pso-csi
 helm repo update
 helm search repo pureStorageDriver -l
```
Comment (pure-jliao) [inline pureStorageDriver/README.md:56]: don't need to mention `--devel` flag here for any rc releases? they are not listed in the result without this flag.

File: pureStorageDriver/README.md (hunk 5 above)
Comment (sdodsley) [reply pureStorageDriver/README.md]: Done

File: pureStorageDriver/README.md (hunk 6 above)
Comment (sdodsley) [reply pureStorageDriver/README.md:56]: Deliberately removed those. 
We shouldn't be advertising them to the world.
If customers are requested to use devel versions we can tell them how to do it - IMHO

**Pull request merged**